tests/integration/targets/inventory/test.yml

```diff
@@ -0,0 +1,25 @@
+- name: Test inventory plugin
+  hosts: localhost
+  gather_facts: false
+
+  tasks:
+    - ansible.builtin.assert:
+        that:
+          - groups | length == 7
+          - "'all' in groups"
+          - "'ungrouped' in groups"
+          - "'childlessgroup' in groups"
+          - "'somegroup' in groups"
+          - "'anothergroup' in groups"
+          - "'somechild' in groups"
+          - "'anotherchild' in groups"
+          - "'somehost' in hostvars"
+          - "'anotherhost' in hostvars"
+          - hostvars["somehost"].hello == "from host!"
+          - hostvars["somehost"].host_variable == "7"
+          - hostvars["anotherhost"].hello == "from anotherhost!"
+          - hostvars["anotherhost"].host_variable == "5"
+          - "'somegroup' in hostvars['somehost']['groups']"
+          - "'anothergroup' in hostvars['somehost']['groups']"
+          - "'somechild' in hostvars['anotherhost']['groups']"
+          - "'ungrouped' in hostvars['ungrupedhost']['groups']"
```

Comment:
Add tests for group variables too please.

Reply:
Added